utils/create_dummy_models.py

```diff
@@ -620,6 +695,87 @@ def build_composite_models(config_class, output_dir):
     return result
 
 
+def get_token_id_from_tokenizer(token_id_name, tokenizer, original_token_id):
+    """Use `tokenizer` to get the values of `bos_token_id`, `eos_token_ids`, etc.
+
+    The argument `token_id_name` should be a string ending with `_token_id`, and `original_token_id` should be an
+    integer that will be return if `tokenizer` has no token corresponding to `token_id_name`.
+    """
+
+    token_id = original_token_id
+
+    if not token_id_name.endswith("_token_id"):
+        raise ValueError(f"`token_id_name` is {token_id_name}, which doesn't end with `_token_id`!")
+
+    token = getattr(tokenizer, token_id_name.replace("_token_id", "_token"), None)
+    if token is not None:
+        if isinstance(tokenizer, PreTrainedTokenizerFast):
+            token_id = tokenizer._convert_token_to_id_with_added_voc(token)
+        else:
+            token_id = tokenizer._convert_token_to_id(token)
+
+    return token_id
+
+
+def get_config_overrides(config_class, processors):
```

Comment:
So far (before this PR), we use the (converted) tokenizer's `vocab_size` to set `tiny_config.vocab_size`.

However, some model testers have `self.xxx_token_id = self.vocab_size - 1`, but we never use the updated `vocab_size` to recreate the model tester, so `self.xxx_token_id` was computed with  the original `vocab_size` (usually `99`).

Also, the special tokens in the tokenizer can have different ids before/after the conversion.

**This method tries its best effort to make sure the `xxx_token_id` in the tiny config (obtained from model tester) will correspond to the (converted) tokenizer, if the corresponding tokens exist in the tokenizer.**
(or at least, respect what has been specified in the model tester's `__init__`)